application/src/Api/Representation/AbstractResourceEntityRepresentation.php

```diff
@@ -501,11 +501,28 @@ public function displaySubjectValues($page = null, $perPage = null, $property =
      * Get the display title for this resource.
      *
      * @param string|null $default
+     * @param string|null $lang
      * @return string|null
      */
-    public function displayTitle($default = null)
+    public function displayTitle($default = null, $lang = null)
     {
-        $title = $this->title();
+        $template = $this->resourceTemplate();
+        if ($template && $template->titleProperty()) {
+            $titleTerm = $template->titleProperty()->term();
+        } else {
+            $titleTerm = 'dcterms:title';
+        }
+
+        if ($lang !== null) {
+            if ($titleValue = $this->value($titleTerm, ['lang' => $lang])) {
+                $title = $titleValue->value();
```

Comment:
In these spots where you're getting a value to return, you should be casting the value object to string: that uses the toString method of the value's data type, which isn't necessarily just returning the `->value()`.

Reply:
OK updated in 81ec357